restore/data.go

```diff
@@ -39,8 +39,7 @@ func CopyTableIn(queryContext context.Context, connectionPool *dbconn.DBConn, ta
 	if singleDataFile || resizeCluster {
 		//helper.go handles compression, so we don't want to set it here
 		customPipeThroughCommand = utils.DefaultPipeThroughProgram
-		errorFile := fmt.Sprintf("%s_error", globalFPInfo.GetSegmentPipePathForCopyCommand())
-		readFromDestinationCommand = fmt.Sprintf("(timeout 300 bash -c \"while [[ ! -p \"%s\" && ! -f \"%s\" ]]; do sleep 1; done\" || (echo \"Pipe not found %s\">&2; exit 1)) && %s", destinationToRead, errorFile, destinationToRead, readFromDestinationCommand)
+		readFromDestinationCommand = fmt.Sprintf("(timeout --foreground 300 bash -c \"while [ ! -p \"%s\" ]; do sleep 1; done\" || (echo \"Pipe not found %s\">&2; exit 1)) && %s", destinationToRead, destinationToRead, readFromDestinationCommand)
```

Comment:
Why should we not check error file during waiting on COPY anymore? There is lack of details in the PR description. How this changes connected with remaining patch?

Also, I strongly recommend to separate adding foreground option to the separate commit with detailed description about timeout behavior.

Reply:
> I strongly recommend to separate adding foreground option to the separate commit with detailed description about timeout behavior.

Unfortunately, neither adding context cancellation to the recover nor adding the foreground argument separately solves the hanging problem.

Reply:
> Why should we not check error file during waiting on COPY anymore? There is lack of details in the PR description. How this changes connected with remaining patch?

I restored checking error files during waiting timeout

Reply:
> > I strongly recommend to separate adding foreground option to the separate commit with detailed description about timeout behavior.
> 
> Unfortunately, neither adding context cancellation to the recover nor adding the foreground argument separately solves the hanging problem.

Maybe split the patch into two commits? The first one will add the `foreground` argument to the `timeout` command, and the second one will add context `cancel` on `recover` and add a new test for it?

Reply:
> Maybe split the patch into two commits?

> I strongly recommend to separate adding foreground option to the separate **commit**

Exactly.